### PR TITLE
Rewrite : CA3075 : DoNotUseInsecureDtdProcessingAnalyzer With Ioperation correct API's

### LINKED ISSUE #1202: Rewrite : CA3075 : DoNotUseInsecureDtdProcessingAnalyzer With Ioperation correct API's
Current [implementation](https://github.com/dotnet/roslyn-analyzers/blob/ce77e9a81de80385aae9e635f9084bc22672e561/src/Microsoft.NetFramework.Analyzers/Core/DoNotUseInsecureDtdProcessing.cs)

Disabled currently as it creates more diagnostics than the unit tests were expecting.
Disabling this to get the Vsix created and unblock the current sceanrio.
Some one will need to look into it, and clean the approach taken

### PR DESCRIPTION
**Pull request by ivanbasov**
Fixes #1202

### CODE REVIEW

File: src/Microsoft.NetFramework.Analyzers/Core/DoNotUseInsecureDtdProcessing.cs [hunk 1]
```diff
@@ -380,18 +378,14 @@ private void AnalyzeObjectCreationForXmlDocument(OperationAnalysisContext contex
             {
                 XmlDocumentEnvironment xmlDocumentEnvironment;
 
-                if (variable == null || !_xmlDocumentEnvironments.ContainsKey(variable))
+                if (variable == null || !_xmlDocumentEnvironments.TryGetValue(variable, out xmlDocumentEnvironment))
```
Comment (333fred) [inline src/Microsoft.NetFramework.Analyzers/Core/DoNotUseInsecureDtdProcessing.cs]: Use out var.

**Comment by 333fred** [comment]
>                             IPropertySymbol prop = context.Compilation.GetSemanticModel(context.Operation.Syntax.SyntaxTree)?.GetSymbolInfo(assign.Target.Syntax).Symbol as IPropertySymbol;

Not related to this PR, but it feels like there should be a better way of doing this. We shouldn't have to back to syntax to get the property symbol of an assignment.

---
Refers to: src/Microsoft.NetFramework.Analyzers/Core/DoNotUseInsecureDtdProcessing.cs:408 in 233118e. [](commit_id = 233118e2c325a4f9ff8ff57b8aec2a6b91669e76, deletion_comment = False)

**Comment by 333fred** [comment]
>                             IPropertySymbol prop = context.Compilation.GetSemanticModel(context.Operation.Syntax.SyntaxTree)?.GetSymbolInfo(assign.Target.Syntax).Symbol as IPropertySymbol;

Perhaps we should add a GetSymbolInfo extension on IOperation? @dotnet/analyzer-ioperation

---
In reply to: [339452143](https://github.com/dotnet/roslyn-analyzers/pull/1327#issuecomment-339452143) [](ancestors = 339452143)

---
Refers to: src/Microsoft.NetFramework.Analyzers/Core/DoNotUseInsecureDtdProcessing.cs:408 in 233118e. [](commit_id = 233118e2c325a4f9ff8ff57b8aec2a6b91669e76, deletion_comment = False)

**Comment by ivanbasov** [comment]
ping @dotnet/analyzer-ioperation

File: src/Microsoft.NetFramework.Analyzers/Core/DoNotUseInsecureDtdProcessing.cs [hunk 2]
```diff
@@ -436,31 +430,13 @@ private void AnalyzeObjectCreationForXmlDocument(OperationAnalysisContext contex
                                 }
                                 else // Non secure resolvers
                                 {
-
-                                    if (operation.Operand is IObjectCreationExpression xmlResolverObjCreated)
```
Comment (mavasani) [inline src/Microsoft.NetFramework.Analyzers/Core/DoNotUseInsecureDtdProcessing.cs]: Tagging @qinxgit , who initially wrote this analyzer and his feedback would be valuable.

**Comment by mavasani** [comment]
@ivanbasov Can you please resolve conflicts and merge?

**Pull request merged**